### PR TITLE
feat(ingest): Add Metadata Ingestion Source Cassandra support

### PR DESCRIPTION
**Pull request by havramar**
Add Metadata Ingestion Source Cassandra support

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Review by kevinhu — COMMENTED** [review]
This is great work! Just a few small nits, but otherwise this looks good and will be useful for a bunch of people.

**Comment by havramar** [comment]
Thank for your review @kevinhu. I have applied your suggestions.

File: metadata-ingestion/src/datahub/ingestion/source/cassandra.py [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+        config = CassandraConfig.parse_obj(config_dict)
+        return cls(ctx,
+                   config)
+
+    def get_workunits(self) -> Iterable[MetadataWorkUnit]:
+        platform = 'cassandra'
+        keyspaces = self._get_db_keyspaces()
+        for keyspace in keyspaces:
+            if not self.config.schema_pattern.allowed(keyspace):
+                print(f'Skipping keyspace: {keyspace}')
```
[the remark this answers is not in the document]
Comment (havramar) [reply metadata-ingestion/src/datahub/ingestion/source/cassandra.py]: Replaced prints with `logger.info`

File: metadata-ingestion/src/datahub/ingestion/source/cassandra.py [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+                             platform: str,
+                             keyspace: str,
+                             table_or_view_type: str,
+                             table: str,
+                             table_schema: cassandra.metadata.TableMetadata) -> Iterable[MetadataWorkUnit]:
+        dataset_name = f'{keyspace}.{table}'
+
+        # 1. Prepare DatasetSnapshot
+        dataset_snapshot = DatasetSnapshot(
+            urn=f'urn:li:dataset:(urn:li:dataPlatform:{platform},{dataset_name},{self.config.env})',
```
[the remark this answers is not in the document]
Comment (havramar) [reply metadata-ingestion/src/datahub/ingestion/source/cassandra.py]: Good catch, applied.

File: metadata-ingestion/tests/integration/cassandra/test_cassandra.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+        # Run the metadata ingestion pipeline.
+        config_file = (test_resources_dir / "cassandra_to_file.yml").resolve()
+        run_datahub_cmd(["ingest", "-c", f"{config_file}"], tmp_path=tmp_path)
+
+        # Verify the output.
+        mce_helpers.check_golden_file(
+            pytestconfig,
+            output_path=tmp_path / "cassandra_mces.json",
+            golden_path=test_resources_dir / "cassandra_cmes_golden.json",
```
[the remark this answers is not in the document]
Comment (havramar) [reply metadata-ingestion/tests/integration/cassandra/test_cassandra.py]: Renamed.

File: metadata-ingestion/src/datahub/ingestion/source/cassandra.py [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+        # 4. Prepare SchemaMetadata
+        actor = 'urn:li:corpuser:etl'
+        sys_time = int(time.time() * 1000)
+        primary_keys = [column_in_primary_key.name for column_in_primary_key in
+                        table_schema.partition_key + table_schema.clustering_key]
+        schema_metadata = SchemaMetadata(
+            schemaName=dataset_name,
+            version=0,
+            hash='',
+            platform=f'urn:li:dataPlatform:{platform}',
```
[the remark this answers is not in the document]
Comment (havramar) [reply metadata-ingestion/src/datahub/ingestion/source/cassandra.py]: Applied.

**Review by kevinhu — APPROVED** [review]

**Comment by jjoyce0510** [comment]
Closing this PR due to inactivity. We are definitely interesting in accepting a Cassandra source, but our framework has changed quite dramatically in the past year and this PR will need to be updated to support.

Thanks
John

**Pull request closed without merging**